### DISCUSSION TITLE
calculate the RGB (ideally 10 bit) values of selected areas of an image

### DISCUSSION THREAD

**Discussion by borneo1910** [ImageMagick/ImageMagick#4970] [Help] [answered]
I want to be able to break up an image into sections by reading the height and width, then dividing it into 10 x 10 area (10x10 for example, but probably some easily divisible numbers, ie a 1920x1080 image into 19 x 10 squares) and then derive the average RGB pixel values of those areas.  Either output as text or as an overlay on the original image.

i can derive the H and W of an image, but I can't find out how to separate the image into the XxY blocks and then calculate the average pixel values.  

Any help would be appreciated

**Comment by fmw42** [comment]
See tile cropping at https://legacy.imagemagick.org/Usage/crop/#crop_tile and the next section

**Reply by borneo1910** [reply]
Thank you.  one thing I';m struggling with is trying to use variables for my crop size, like so:

convert $theSelectedFile -crop '$cropWidth'x'$cropHeight' $theSelctedFile"_crop"

I'm sure I just don't have the right formatting, but I can't find any info on the correct formatting.

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform?

> I can't find out how to separate the image into the XxY blocks ...

If you want to split the image into blocks of 10x10 pixels, then "-crop 10x10 +repage". In general, with IM v7, you can use %[fx...] expressions instead of numbers.

> ... and then calculate the average pixel values.

"-scale 1x1!" will average each image down to one pixel.

So, for example, assuming your IM version is recent enough to include "ftxt:"

```
f:\web\im>%FXNEW%magick rose: -crop 10x10 +repage -scale "1x1^!" -define ftxt:format="\c\n" ftxt:

9.0766976e+08,8.5276155e+08,8.0240095e+08
9.1120679e+08,8.2210727e+08,7.5321936e+08
1.487743e+09,9.7352592e+08,8.3423424e+08
2.8114351e+09,1.0584147e+09,8.7617333e+08
2.6005606e+09,1.3919063e+09,1.187769e+09
1.4052122e+09,1.0137807e+09,9.9036893e+08
1.9227979e+09,2.0839855e+09,1.9007336e+09
1.3812952e+09,1.2522777e+09,1.1289869e+09
2.3592003e+09,1.7014808e+09,2.1848751e+09
3.0509427e+09,1.0501616e+09,9.157544e+08
3.7499275e+09,8.9234262e+08,8.1789652e+08
3.5060408e+09,1.0481405e+09,8.8644756e+08
3.8439115e+09,1.4931327e+09,1.6592048e+09
3.5021669e+09,3.5631386e+09,3.2840499e+09
1.5999174e+09,1.4933012e+09,1.6969332e+09
2.7738752e+09,1.9404831e+09,2.1811697e+09
3.586045e+09,1.046793e+09,8.6556223e+08
3.8083728e+09,1.0338239e+09,1.0508353e+09
3.7605386e+09,9.470824e+08,8.7196258e+08
3.1120828e+09,1.2578359e+09,1.2175811e+09
1.8464991e+09,2.4294356e+09,1.4008331e+09
1.7636315e+09,1.6226555e+09,1.3743895e+09
2.3162506e+09,1.8781639e+09,1.6666157e+09
3.4310894e+09,1.3209972e+09,1.1948431e+09
2.7097033e+09,1.0043486e+09,8.3575011e+08
2.3570107e+09,9.5415646e+08,7.8016818e+08
1.5557887e+09,1.3199866e+09,1.0257392e+09
1.541809e+09,2.0363198e+09,1.1739577e+09
1.7746917e+09,1.7508308e+09,1.5088529e+09
3.8587334e+09,3.866032e+09,3.7310072e+09
4.147591e+09,4.0072326e+09,3.9547385e+09
2.52982e+09,1.8007984e+09,1.5874536e+09
1.5445039e+09,1.2851216e+09,9.5696363e+08
8.7050285e+08,9.6285868e+08,8.2193884e+08
9.7998241e+08,1.2059594e+09,9.1092607e+08
```
This gives the values in a nominal range of zero to QuantumRange, which in my case is 2^32-1. If you want a scale of 0 to 255:
```
f:\web\im>%FXNEW%magick rose: -crop 10x10 +repage -scale "1x1^!" -evaluate Multiply %[fx:255/QuantumRange] -define ftxt:format="\c\n" ftxt:
53.89,50.63,47.64
54.1,48.81,44.72
88.33,57.8,49.53
166.92,62.84,52.02
154.4,82.64,70.52
83.43,60.19,58.8
114.16,123.73,112.85
82.01,74.35,67.03
140.07,101.02,129.72
181.14,62.35,54.37
222.64,52.98,48.56
208.16,62.23,52.63
228.22,88.65,98.51
207.93,211.55,194.98
94.99,88.66,100.75
164.69,115.21,129.5
212.91,62.15,51.39
226.11,61.38,62.39
223.27,56.23,51.77
184.77,74.68,72.29
109.63,144.24,83.17
104.71,96.34,81.6
137.52,111.51,98.95
203.71,78.43,70.94
160.88,59.63,49.62
139.94,56.65,46.32
92.37,78.37,60.9
91.54,120.9,69.7
105.36667,103.95,89.583333
229.1,229.53333,221.51667
246.25,237.91667,234.8
150.2,106.91667,94.25
91.7,76.3,56.816667
51.683333,57.166667,48.8
58.183333,71.6,54.083333
```

**Reply by borneo1910** [reply]
Thank you very much!  this is exactly what I was looking for.  One step closer to what I'm trying to achieve.  I owe you a beer.  

Love this community.
